Deployment/terraform_conditional_run.ps1

```diff
@@ -0,0 +1,53 @@
+param (
+    [Parameter(Mandatory = $true)] [string] $deploymentResourceGroupName,
+    [Parameter(Mandatory = $true)] [string] $deploymentStorageAccountName,
+    [Parameter(Mandatory = $true)] [string] $workSpace,
+    [Parameter(Mandatory = $true)] [boolean] $continueEvenIfResourcesAreGettingDestroyed
+)
+
+cd $env:AGENT_BUILDDIRECTORY/terraformartifact/src
+
+Write-output "Executing terraform scripts for deployment in $workSpace enviroment"
+terraform init -backend-config="resource_group_name=$deploymentResourceGroupName" -backend-config="storage_account_name=$deploymentStorageAccountName" -backend-config="key=terraform.deployment.tfplan"
+if ( !$? ) { echo "Something went wrong during terraform initialization"; throw "Error" }
+
+Write-output "Selecting workspace"
+
+$ErrorActionPreference = 'SilentlyContinue'
+terraform workspace new $WorkSpace 2>&1 > $null
+$ErrorActionPreference = 'Continue'
+
+terraform workspace select $workSpace
+if ( !$? ) { echo "Error while selecting workspace"; throw "Error" }
+
+Write-output "Validating terraform"
+terraform validate
+if ( !$? ) { echo "Something went wrong during terraform validation" ; throw "Error" }
+
+Write-output "Execute Terraform plan"
+terraform plan -out "terraform.deployment.tfplan" -no-color | tee terraform_output.txt
```

Comment:
Is there a reason for no colour? It helps us identify additions and removals much more easily

Reply:
Removed it.

Reply:
Script is uplifted from fss looks like it was used there as we are writing output to file. I removed and tried running it and it works